### PR TITLE
Add a top-level class for metrics similar with all the other packages.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/openconsensus/metrics/ToDoubleFunction.java [hunk 1]
```diff
@@ -14,12 +14,10 @@
  * limitations under the License.
  */
 
-package openconsensus.common;
-
-import openconsensus.metrics.MetricRegistry;
+package openconsensus.metrics;
```
Comment (songy23) [inline api/src/main/java/openconsensus/metrics/ToDoubleFunction.java]: OC-Shim doesn't need to be updated.

File: opencensus-shim/src/main/java/openconsensus/opencensusshim/metrics/Metrics.java [hunk 2]
```diff
@@ -48,10 +48,10 @@ public static ExportComponent getExportComponent() {
   /**
    * Returns the global {@link MetricRegistry}.
    *
-   * <p>This {@code MetricRegistry} is already added to the global {@link
+   * <p>This {@code MetricCollection} is already added to the global {@link
```
Comment (songy23) [inline opencensus-shim/src/main/java/openconsensus/opencensusshim/metrics/Metrics.java]: OC-Shim doesn't need to be updated. (previous comment is under wrong file)

Comment (bogdandrutu) [reply opencensus-shim/src/main/java/openconsensus/opencensusshim/metrics/Metrics.java]: Done

File: api/src/main/java/openconsensus/metrics/DerivedDoubleCumulative.java [hunk 3]
```diff
@@ -30,12 +29,13 @@
...
  * class YourClass {
  *
- *   private static final MetricRegistry metricRegistry = Metrics.getMetricRegistry();
+ *   private static final MetricCollection metricCollection =
+ *       Metrics.getMeter().createMetricsCollection().build();
  *
  *   List<LabelKey> labelKeys = Arrays.asList(LabelKey.create("Name", "desc"));
  *   List<LabelValue> labelValues = Arrays.asList(LabelValue.create("Inbound"));
  *
- *   DerivedDoubleCumulative cumulative = metricRegistry.addDerivedDoubleCumulative(
```
Comment (SergeyKanzhelev) [inline api/src/main/java/openconsensus/metrics/DerivedDoubleCumulative.java]: I like registry more. What's the rationale for renaming?

File: api/src/main/java/openconsensus/metrics/MetricCollection.java [hunk 4]
```diff
@@ -104,4 +103,37 @@
...
+    /**
+     * Sets the name of the component that reports these metrics.
+     *
+     * <p>The final name of the reported metric will be <code>component + "_" + name</code> if the
+     * component is not empty.
+     *
+     * @param component the name of the component that reports these metrics.
+     * @return this.
+     */
+    public abstract Builder setComponent(String component);
```
Comment (SergeyKanzhelev) [inline api/src/main/java/openconsensus/metrics/MetricCollection.java]: hm... is there any relationship of `component` and `Resource`?

Comment (SergeyKanzhelev) [reply api/src/main/java/openconsensus/metrics/MetricCollection.java]: perhaps mention it in summary as minimum

Comment (bogdandrutu) [reply api/src/main/java/openconsensus/metrics/MetricCollection.java]: * Component: will be something like "apache_kafka" - it is important when do instrumentation.
* Resource: it is used when reporting out-of-band metrics, for example a sidecar reading logs and generating metrics on behalf of the task that generated the logs.

File: api/src/main/java/openconsensus/metrics/DerivedDoubleCumulative.java (hunk 3 above)
Comment (bogdandrutu) [reply api/src/main/java/openconsensus/metrics/DerivedDoubleCumulative.java]: I can keep the name registry. The only concern that I had was that Registry sounds a bit more global than collection. I would go with registry for the moment.

**Comment by bogdandrutu** [comment]
@SergeyKanzhelev I reverted the class name change, fix some warnings in the tests as well (unused variables).

**Pull request merged**